src/TLuaInterpreter.cpp

```diff
@@ -3307,35 +3106,18 @@ int TLuaInterpreter::getFontSize(lua_State* L)
 int TLuaInterpreter::openUserWindow(lua_State* L)
 {
     int n = lua_gettop(L);
-    if (lua_type(L, 1) != LUA_TSTRING) {
-        lua_pushfstring(L, "openUserWindow:  bad argument #1 type (name as string expected, got %s!)", luaL_typename(L, 1));
-        return lua_error(L);
-    }
-    QString name{lua_tostring(L, 1)};
+    QString name = getVerifiedString(L, __func__, 1, "name");
```

Comment:
:warning: `getVerifiedString(...)` uses `lua_isstring(...)` however that will not behave the same as a test for:
`(lua_type(L, 1) != LUA_TSTRING)`
because, due to Lua's type coercion capabilities `lua_isstring(...)` will return `true` if the argument is also a **number** whereas the other test will not.

Reply:
Good comment. Need to review.
There are more than a few dozen different ways we do argument testing throughout TLuaInterpreter. Need to find some best practice and stick to it as often as possible.

Reply:
Is this a deliberate decision here, or could it be adjusted?
From what I am seeing, currently this openUserWindow function you comment on will throw an error if I try `openUserWindow(5)` as this is not "real" string it will not let me create that window. If I want a user window with title 5, I need to instead put `openUserWindow("5")` and that will work fine.
I would like to reason, this is actually a change for the better.
If there are any deliberate choices of LUA_TSTRING as opposed to lua_isstring, please do comment on those. I would love to rewire them to use lua_isstring instead, but if it can't be done I may need to backtrack.

Reply:
It's good to follow the Lua standard in coercion

Reply:
> It's good to follow the Lua standard in coercion

I did not understand what you are suggesting here.

Reply:
I assume it means: revert my changes here, then open a new issue to upgrade these functions to the (which?) Lua coercion standard, so maybe SlySven could comply, before this refactoring can happen

Reply:
If I get @SlySven right he's saying not to do the coercion, I think we should follow Lua here in allowing coercion. That is, do `lua_isstring()` and instead of `(lua_type(L, 1) == LUA_TSTRING)`

Reply:
Yes he (or somebody else?) originally deliberately put LUA_TSTRING and not lua_isstring here.
Now my original change to getVerifiedString would change this 1:1 to use lua_isstring.
That seems not wanted by SlySven, probably because unexpected logic changes.
So the original author or somebody must change logic to use lua_isstring.

Reply:
I have undone these changes in the 5 affected functions using LUA_T* for argument type checks.

There are 50+ uses of LUA_T* in this file, so probably another undertaking to update them all without breaking logic.

Would you like to file an issue for this updating logic to use coercion? I can't judge if 1:1 is possible without side-effects.

Reply:
> I have undone these changes in the 5 affected functions using LUA_T* for argument type checks.

Then re-done some changes that improve readability, but did not touch coercion this time in those places.

Reply:
After looking at these 5 functions more, I see no reason why LUA_TSTRING needs to be kept over lua_isstring.
@SlySven please elaborate how they can't be converted in these 5 cases right now.